runtime/auth/aws-signing-default/common/src/aws/smithy/kotlin/runtime/auth/awssigning/RequestMutator.kt

```diff
@@ -49,14 +47,14 @@ internal class DefaultRequestMutator : RequestMutator {
             }
 
             AwsSignatureType.HTTP_REQUEST_VIA_QUERY_PARAMS -> {
-                with(canonical.request.url.parameters) {
-                    set("X-Amz-Signature", signatureHex)
+                canonical.request.url.parameters.decodedParameters.put("X-Amz-Signature", signatureHex)
 
-                    entries().forEach {
-                        remove(it.key)
-                        appendAll(it.key, it.value.map(String::urlReencodeComponent))
-                    }
+                /* TODO don't need to reencode because their already canonicalized by `Encodable`?
```

Comment:
did we answer this?

Reply:
My signing tests verified that we _do_ need to reencode in the request mutator. I've already updated the code in this class and I'll remove the TODO.